### ISSUE TITLE
Implement nohl to disable all search highlights

### ISSUE THREAD

**Issue by FSMaxB** [VSCodeVim/Vim#1047]
Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----

### What did you do?
Search for a word contained in the open file with `/`. Then typing `:nohl`.

### What did you expect to happen?

The search highlight for the current search should be disabled.

### What happened instead?

`:nohl` had no effect.

### Technical details:

* VSCode Version: 1.7.1
* VsCodeVim Version: 0.4.1
* OS:  Windows 8.1

**Comment by rebornix** [comment]
Interesting, currently for configurations we don't have alias support but it should be easy.

**Comment by FSMaxB** [comment]
`:nohlsearch` doesn't work either, so I don't know how his is related to aliasing. This is not about changing the settings in `settings.json`, these should not be touched!

From the vim help:
```
							*:noh* *:nohlsearch*
:noh[lsearch]		Stop the highlighting for the 'hlsearch' option.  It
			is automatically turned back on when using a search
			command, or setting the 'hlsearch' option.
			This command doesn't work in an autocommand, because
			the highlighting state is saved and restored when
			executing autocommands |autocmd-searchpat|.
			Same thing for when invoking a user function.
```

**Comment by rebornix** [comment]
@FSMaxB are you using Insider as well? It might share the same root cause as https://github.com/VSCodeVim/Vim/issues/423#issuecomment-259600175 .

**Comment by FSMaxB** [comment]
No, I'm not using insider. How is this related to #423?

**Comment by rebornix** [comment]
@FSMaxB the reason that #423 doesn't work is that setting relative line number is actually changing Code's one config. Thanks for the help!

Now this one is way more severe than I thought.

**Comment by FSMaxB** [comment]
I wouldn't consider the priority critical, but it's definitely a nice to have feature.

Workaround: Make a new empty search `/<enter>`. That has almost the same effect (except that the search term gets lost).

**Comment by johnfn** [comment]
In the meantime you can put `vim.hlsearch` in your settings.json, which has the same effect.

**Comment by FSMaxB** [comment]
No, it does not have the same effect. Setting `vim.hlsearch` to `false` disables the highlighting completely. That is not what `:nohlsearch` does!

**Comment by FSMaxB** [comment]
>Stop the highlighting for the 'hlsearch' option.  It is **automatically turned back on** when using a search command

from the vim help

**Comment by rebornix** [comment]
`set nohlsearch` now works. But alias is still not supported.

**Comment by rebornix** [comment]
Sorry, close it too quick. It's about `:nohl` but not disable it globally.

**Comment by johnfn** [comment]
Added in master and will be released in 0.4.7!

**Comment by FSMaxB** [comment]
Works great, thanks!
